druid-lookups/src/main/java/com/yahoo/maha/maha_druid_lookups/server/lookup/namespace/JDBCExtractionNamespaceCacheFactory.java

```diff
@@ -96,73 +89,146 @@ Void populateRowListFromJDBC(
         }
 
         List<Void> p = handle.createQuery(query).map(rm)
                 .setFetchSize(FETCH_SIZE)
-                .bind("lastUpdatedTimeStamp", updateTS)
+                .bind("lastUpdatedTimeStamp",
+                        getTsValue(extractionNamespace, updateTS))
                 .list();
 
         return null;
     }
 
+    protected String getWhereClauseExtension(String id, JDBCExtractionNamespace extractionNamespace) {
+        String whereClauseExtension = "";
+        if (extractionNamespace.hasTsColumnConfig() && extractionNamespace.getTsColumnConfig().hasSecondaryTsColumn()) {
+            String maxVal = (String) getMaxValFromColumn(id, extractionNamespace, StringMapper.FIRST, extractionNamespace.getTsColumnConfig().getSecondaryTsColumn(), extractionNamespace.getTable());
+            whereClauseExtension = String.format(WHERE_CLAUSE_EXTENSION,
+                    extractionNamespace.getTsColumnConfig().getSecondaryTsColumn(),
+                    extractionNamespace.getTsColumnConfig().getSecondaryTsColumnCondition(),
+                    StringUtil.quoteStringLiteral(maxVal)
+            );
+        }
+        return whereClauseExtension;
+    }
+
+    private Object getTsValue(JDBCExtractionNamespace extractionNamespace, Timestamp updateTS) {
+        Object tsValue = updateTS;
+        if (extractionNamespace.hasTsColumnConfig()) {
+            if (extractionNamespace.getTsColumnConfig().isVarchar()) {
+                tsValue = new SimpleDateFormat(extractionNamespace.getTsColumnConfig().getFormat()).format(updateTS);
+            } else if (extractionNamespace.getTsColumnConfig().isBigint()) {
+                tsValue = updateTS.getTime();
+            }
+        }
+        return tsValue;
+    }
+
+    protected String getBaseWhereClause(String baseClause, JDBCExtractionNamespace namespace) {
+        return String.format(
+                baseClause,
+                namespace.getTsColumn(),
+                namespace.hasTsColumnConfig()?
+                        (namespace.getTsColumnConfig().isVarchar() ?
+                                StringUtil.quoteStringLiteral(LAST_UPDATED_TIMESTAMP)
+                                : LAST_UPDATED_TIMESTAMP)
+                        : LAST_UPDATED_TIMESTAMP
+        );
+    }
+
     protected DBI ensureDBI(String id, JDBCExtractionNamespace namespace) {
         final String key = id;
         DBI dbi = null;
+
         if (dbiCache.containsKey(key)) {
             dbi = dbiCache.get(key);
         }
+
         if (dbi == null) {
-            final DBI newDbi = new DBI(
-                    namespace.getConnectorConfig().getConnectURI(),
-                    namespace.getConnectorConfig().getUser(),
-                    namespace.getConnectorConfig().getPassword()
-            );
-            dbiCache.putIfAbsent(key, newDbi);
-            dbi = dbiCache.get(key);
+            if (!namespace.hasKerberosProperties()) {
```

Comment:
hasKerberosProperties checks for size != 0, so even if the json is empty "{}", this logic will attempt to connect with kerberos. We can either have an explicit config to say which auth should be used (userid/password or kerberos) or enhance the check for 'hasKerberosProperties'.

Reply:
if the json is empty "{}", the size of `KerberosProperties` would be 0.. However, having an explicit config sounds a better idea. Will add another field for that.